tortoisebeacon/handlers.go

```diff
@@ -101,37 +103,37 @@ func (tb *TortoiseBeacon) handleProposalMessage(ctx context.Context, m ProposalM
 	currentEpoch := tb.currentEpoch()
 
 	atxID, err := tb.verifyProposalMessage(ctx, m, currentEpoch)
-	if errors.Is(err, ErrProposalDoesntPassThreshold) || errors.Is(err, database.ErrNotFound) {
-		return nil
-	}
-
 	if err != nil {
 		return err
 	}
 
-	if err := tb.classifyProposalMessage(ctx, m, atxID, currentEpoch, receivedTime); err != nil {
-		return fmt.Errorf("classify proposal message: %w", err)
+	if err = tb.classifyProposalMessage(ctx, m, atxID, currentEpoch, receivedTime); err != nil {
+		return err
 	}
 
 	return nil
 }
 
 func (tb *TortoiseBeacon) classifyProposalMessage(ctx context.Context, m ProposalMessage, atxID types.ATXID, currentEpoch types.EpochID, receivedTime time.Time) error {
+	minerID := m.NodeID.ShortString()
+	logger := tb.logger.WithContext(ctx).WithFields(currentEpoch, log.String("miner_id", minerID))
+
 	atxHeader, err := tb.atxDB.GetAtxHeader(atxID)
 	if err != nil {
-		return fmt.Errorf("failed to get ATXID %v header: %w", atxID, err)
+		logger.Error("[proposal] failed to get ATX header", atxID, log.Err(err))
+		return fmt.Errorf("[proposal] failed to get ATX header (miner ID %v, ATX ID %v): %w", minerID, atxID, err)
 	}
 
 	atxTimestamp, err := tb.atxDB.GetAtxTimestamp(atxID)
 	if err != nil {
-		return fmt.Errorf("failed to get ATXID %v timestamp: %w", atxID, err)
+		logger.Error("[proposal] failed to get ATX timestamp", atxID, log.Err(err))
```

Comment:
same

Reply:
same here. if we didn't save ATX timestamp, that's a programming error